pandas/io/stata.py

```diff
@@ -1494,7 +1491,11 @@ def read(self, nrows=None, convert_dates=None,
             data = data.set_index(ix)
 
         if columns is not None:
-            data = self._do_select_columns(data, columns)
+            try:
```

Comment:
Yeah, that was my thought as soon as I saw your previous comment. I'm open to implementing it if you want it but, as you say, it's probably not worth it.